pkg/providers/vsphere/vsphere.go

```diff
@@ -443,6 +450,80 @@ func (p *vsphereProvider) validateMachineConfigsNameUniqueness(ctx context.Conte
 	return nil
 }
 
+func (p *vsphereProvider) getCurrentMachineConfigDatastoreUsage(ctx context.Context, machineConfig *v1alpha1.VSphereMachineConfig, vsphereClusterSpec *Spec, cluster *types.Cluster) (float64, error) {
```

Comment:
Its helpful for maintainers if we indicate the unit being returned via the signature. Some options: create a return parameter `unitGiB` or document it in a method comment.